src/gui/processing/models/qgsmodelviewtoollink.cpp

```diff
@@ -196,32 +231,45 @@ void QgsModelViewToolLink::deactivate()
 void QgsModelViewToolLink::setFromSocket( QgsModelDesignerSocketGraphicItem *socket )
 {
   mFromSocket = socket;
+  mPreviousInputChildId.clear();
+  mPreviousInputSocketNumber = -1;
 
+  // If it's an input socket and it's already connected, we want 'From' to be the output at the other end of the connection
   if ( mFromSocket->isInput() )
   {
+    mPreviousInputSocketNumber = mFromSocket->index();
```

Comment:
it would be tidier if this is set at together with mPreviousInputChildId so that they are consistently set